MultiLangProgGuide/MultiLangProgGuide.rst

```diff
@@ -899,12 +903,75 @@ ydb_lock_s()
 Note that the parameter list **must** be terminated by a NULL pointer.
 
 Release any locks held by the process, attempt to acquire all the
-specified locks. While the release is unconditional, on return, the
-function will have acquired all specified locks or no specified
-locks. If no locks are specified, the function releases all locks and
-returns ``YDB_OK``.
+requested locks. While the release is unconditional, on return, the
+function will have acquired all requested locks or none of them. If no
+locks are requested, the function releases all locks and returns
+``YDB_OK``.
 
-``timeout`` specifies a time in
+``timeout`` specifies a time in microseconds that the function waits
```

Comment:
microseconds? Timers are in milli-seconds. We can obviously convert but unless you are going to have the milli-second to micro-second project happen at some point, a microsecond argument is not very useful. Also an integer microsecond value nstead of a long makes a microsecond argument fairly restrictive. It might should be "long long" so the field is 64 bits even on 32 bit platforms. The timeout value needs to be signed so we can pass NO_TIMEOUT as -1 where it waits forever.

Reply:
Narayanan and I discussed it, and agreed that it was OK for the Simple API to require timeouts and not support untimed locks. While this does not prevent hangs, it does prevent deadlocks. So, unsigned for the timeout is fine.

Since the underlying operating system timers are in microseconds, and a millisecond is a very long time on a computer system, let's use microseconds in the timeout. While the M side uses milliseconds, there's no reason for the C side to be so limited (and sub-millisecond timers would then require floating point).

With a 32-bit value, a time in microseconds is a bit over 1 hour, which effectively limits the duration of a hang. With a 64-bit value, a timeout and a hang would be indistinguishable.

Reply:
Just to be sure there is no misunderstanding, the timeout parameter to the underlying functions is a signed integer in milliseconds. So we:

1. Have to convert microseconds to milliseconds.
1. Have to convert from unsigned to signed

To be clear - there is no micro-second timer support at this time. That is a non-trivial project outside the scope of this project.

Reply:
As discussed, this will be unsigned long long with a symbolic constant for the maximum supported timeout.